### PR TITLE
fixing execution order for dynamic data stream creation

### PR DESCRIPTION
**Pull request by applike-ss**
Should fix partial issue in #925

(check all that apply)
- [ ] tests added
- [x] tests passing
- [ ] README updated (if needed)
- [ ] README Table of Contents updated (if needed)
- [x] History.md and `version` in gemspec are untouched
- [x] backward compatible
- [ ] feature works in `elasticsearch_dynamic` (not required but recommended)

I will first test this in our test environment before ticking either of the boxes.

### CODE REVIEW

**Comment by applike-ss** [comment]
i let the tests run locally and while it did complain that it couldn't reach an elasticsearch node (because i didn't have any running in my docker container that ran the test). Though the tests say it 100% passed :see_no_evil:

**Comment by applike-ss** [comment]
@fiscafusca Mind a review + merge? :-)

**Comment by fiscafusca** [comment]
@applike-ss I would but I'm just a contributor! You need to wait for one of the maintainers to merge your PR :)
Anyway, the fix is just fine 👍🏻

**Comment by applike-ss** [comment]
Ahh somehow i got into thinking this is your project :see_no_evil:

**Comment by cosmo0920** [comment]
> i let the tests run locally and while it did complain that it couldn't reach an elasticsearch node (because i didn't have any running in my docker container that ran the test). Though the tests say it 100% passed 🙈

This is because this Elasticsearch plugin's testcases use webmock to mimic HTTP requests by mocking code.
It is expected behavior.

**Review by cosmo0920 — APPROVED** [review]
Looks good. Thanks for your contribution! :+1:

**Comment by cosmo0920** [comment]
Yay, you'd succeeded to become a contributor, @applike-ss !

**Pull request merged**
